### ISSUE TITLE
Bug: The favicon() function should return a moodle_url object instead of a string

### ISSUE THREAD

**Issue by abias** [moodle-an-hochschulen/moodle-theme_boost_union#130]
In https://github.com/moodle-an-hochschulen/moodle-theme_boost_campus/issues/120, an issue was raised for Boost Campus. It pointed to https://tracker.moodle.org/browse/MDL-66272 which made clear that the `favicon()` function should return a `moodle_url` object instead of a string.

In Boost Union, this function currently looks like this:

```
    public function favicon() {
        if (!empty($this->page->theme->settings->favicon)) {
            return $this->page->theme->setting_file_url('favicon', 'favicon');
        } else {
            return $this->image_url('favicon', 'theme');
        }
    }
```

While the `image_url()` function returns a `moodle_url` object, the `setting_file_url()` function does not.

We will rebuild the favicon() function to the way how it is build in Moodle 4.1dev core (where the custom favicon is introduced into the Moodle core).

**Comment by lucaboesch** [comment]
Closed by #131. Does that fix #78 too (or make it obsolete)?

**Comment by abias** [comment]
> Does that fix https://github.com/moodle-an-hochschulen/moodle-theme_boost_union/issues/78 too (or make it obsolete)?

@lucaboesch - good question. But no, is doesn't. In #78 , we will be able to remove the favicon setting from Boost Union together with the favicon() renderer function completely.
